kokain-processor/src/main/java/com/schwarz/kokain/processor/model/EFactoryModel.kt

```diff
@@ -1,38 +1,32 @@
 package com.schwarz.kokain.processor.model
 
 import com.schwarz.kokain.api.EFactory
-import com.squareup.kotlinpoet.ClassName
+import com.schwarz.kokain.kokaingeneratorlib.model.IEFactoryModel
 import com.squareup.kotlinpoet.TypeName
+import com.squareup.kotlinpoet.asTypeName
 import com.sun.tools.javac.code.Symbol
 import javax.lang.model.element.Element
 import javax.lang.model.type.MirroredTypesException
 import javax.lang.model.type.TypeMirror
 import javax.lang.model.util.Elements
 
-class EFactoryModel(sourceElement: Element, util: Elements) {
+class EFactoryModel(sourceElement: Element, util: Elements) : IEFactoryModel {
 
-    var additionalFactories: Array<TypeMirror>
+    var additionalFactoriesTypeMirror: Array<TypeMirror> = sourceElement.getAnnotation(EFactory::class.java).let {
+        try {
+            it.additionalFactories.map { util.getTypeElement(it.qualifiedName).asType() }
+                .toTypedArray() // this should throw
```

Comment:
I guess, "this could throw" instead of "this should throw"? I hope it usually does not throw :-)

Reply:
na I think it should really throw its some ugly workaround for the old java annotation processor :D

Reply:
haha ok

Before, if I understood it correctly, this was in a method which could or could not be called.
But now, this is always executed as part of the constructor, right? In that case it is not very nice if we throw (and catch) an exception, every time we instantiate the class.